### PR TITLE
added failover strategy

### PR DESCRIPTION
**Pull request by bartes**

### CODE REVIEW

File: lib/castle/fake_auth_response.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Castle
+  # generate fake authentication response
+  class FakeAuthResponse
```
Comment (lluft) [inline lib/castle/fake_auth_response.rb]: @bartes I'd prefer calling this `FailOverAuthRespsonse`

**Comment by bartes** [comment]
TODO: added readme and changelog information

File: lib/castle/configuration.rb [hunk 2]
```diff
@@ -57,6 +59,12 @@ def valid?
       !api_secret.to_s.empty? && !api_endpoint.scheme.nil?
     end
 
+    def failover_strategy=(value)
+      @failover_strategy = FAILOVER_STRATEGIES.detect { |strategy| strategy == value.to_sym }
+      raise Castle::ConfigurationError, 'there is no such a strategy' if @failover_strategy.nil?
```
Comment (lluft) [inline lib/castle/configuration.rb]: change the error message to `unrecognized failover strategy`.

**Comment by bartes** [comment]
@lluft - done but this PR is build on the top of context object so it can be merged  to https://github.com/castle/castle-ruby/pull/46 or to the master when contexts objects are merged there.

**Pull request merged**